cmd/collector/app/handler.go

```diff
@@ -80,7 +87,45 @@ func (aH *APIHandler) saveSpan(w http.ResponseWriter, r *http.Request) {
 		} else {
 			w.WriteHeader(http.StatusOK)
 		}
+
+	case "zipkin.thrift":
+		spans, err := deserializeZipkin(bodyBytes)
+		if err != nil {
+			http.Error(w, fmt.Sprintf(unableToReadBodyErrFormat, err), http.StatusBadRequest)
+			return
+		}
+
+		ctx, _ := tchanThrift.NewContext(time.Minute)
+		if _, err = aH.zipkinSpansHandler.SubmitZipkinBatch(ctx, spans); err != nil {
+			http.Error(w, fmt.Sprintf("Cannot submit Zipkin batch due to error: %v", err), http.StatusInternalServerError)
+			return
+		}
+
+		w.WriteHeader(http.StatusOK)
+
 	default:
 		http.Error(w, fmt.Sprintf("Unsupported format type: %v", format), http.StatusBadRequest)
 	}
 }
+
+func deserializeZipkin(b []byte) ([]*zipkincore.Span, error) {
+	buffer := thrift.NewTMemoryBuffer()
+	buffer.Write(b)
+
+	transport := thrift.NewTBinaryProtocolTransport(buffer)
+	_, size, err := transport.ReadListBegin()
+	if err != nil {
+		return nil, err
+	}
+
+	var spans []*zipkincore.Span
+	for i := 0; i < size; i++ {
+		zs := &zipkincore.Span{}
+		if err = zs.Read(transport); err != nil {
+			return nil, err
+		}
+		spans = append(spans, zs)
+	}
```

Comment:
I did not call EndReadList because it is unnecessary. 
The code in the current thrift implementation does nothing
```
func (p *TBinaryProtocol) ReadListEnd() error {
	return nil
}
```